### PR TITLE
Add pooling and softmax unittests for Arm backend

### PR DESCRIPTION
**Pull request by oscarandersson8218**
* Add SoftMax unittests
* Add mean_dim unittests
* Add AvgPool2d unittests
* Enable linear u55 test

### CODE REVIEW

File: backends/arm/test/test_models.py [hunk 1]
```diff
@@ -221,23 +180,7 @@ def forward(self, x):
...
-
-        def __init__(self):
-            super().__init__()
-            self.softmax = torch.nn.Softmax(dim=1)
-
-        def forward(self, x):
-            return self.softmax(x)
-
-    @register_test
+    # @register_test
```
Comment (digantdesai) [inline backends/arm/test/test_models.py:183]: Is this intentional?

**Review by digantdesai — APPROVED** [review]

File: backends/arm/test/test_models.py (hunk 1 above)
Comment (oscarandersson8218) [reply backends/arm/test/test_models.py]: No, pushed a new patch reverting that change.

**Comment by digantdesai** [comment]
Rebase please?

**Review by mergennachin — APPROVED** [review]
